Dockerfile

```diff
@@ -6,20 +6,31 @@
 #
 # SPDX-License-Identifier: EPL-2.0
 #
+
+# Builder: check meta.yamls and create index.json
 FROM alpine:3.10 AS builder
-RUN apk add --no-cache py-pip jq bash && pip install yq jsonschema
+RUN apk add --no-cache py-pip jq bash wget && pip install yq jsonschema
 
-COPY .htaccess README.md ./scripts/*.sh ./scripts/meta.yaml.schema /build/
+COPY ./scripts/*.sh ./scripts/meta.yaml.schema /build/
 COPY /v3 /build/v3
 WORKDIR /build/
 RUN ./check_plugins_location.sh v3
 RUN ./set_plugin_dates.sh v3
 RUN ./check_plugins_viewer_mandatory_fields.sh v3
 RUN ./ensure_latest_exists.sh
 RUN ./index.sh v3 > /build/v3/plugins/index.json
+RUN chmod -R g+rwX /build
+
+# Build registry, copying meta.yamls and index.json from builder
+FROM registry.centos.org/centos/httpd-24-centos7 AS registry
+COPY README.md .htaccess /var/www/html/
+COPY --from=builder /build/v3 /var/www/html/v3
+
+# Offline build: cache .theia and .vsix files in registry itself and update metas
+FROM builder AS offline-builder
+RUN ./cache_artifacts.sh v3
```

Comment:
It's confusing reading the dockerfile to have `chmod` in the builder and not in the offline-builder. I prefer it as a layer here.